plugins/kubernetes/app/assets/javascripts/kubernetes/controllers/kubernetes_dashboard_ctrl.js

```diff
@@ -1,15 +1,184 @@
-samson.controller('KubernetesDashboardCtrl', function($scope, $stateParams, SseFactory) {
-  $scope.project_id = $stateParams.project_id;
-  $scope.messages = [];
-
-  function init() {
-    // Subscribe to the SSE channel for K8s
-    SseFactory.on('k8s', function(msg) {
-      console.log("Got SSE msg: ", msg);
-      $scope.messages.push(JSON.stringify(msg));
-      _.defer(function() { $scope.$apply(); });
-    });
-  }
-
-  init();
-});
+samson.controller('KubernetesDashboardCtrl',
+  function($rootScope, $scope, $stateParams, $q, environmentsService, kubernetesService, SseFactory, notificationService) {
+    $scope.project_id = $stateParams.project_id;
+    $scope.environments = [];
+    $scope.messages = [];
+    $scope.dashboard_data = [];
+    $scope.environment = undefined;
+    $scope.loadingEnvironments = false;
+    $scope.loadingClusterState = false;
+
+    $scope.reloadDashboardData = function() {
```

Comment:
Hummm, I don't remember why did I create this extra method. I think I was also re-registering the callback for the SSE events in here before realising that was not needed. 

I'm going to simplify this.